### PR TITLE
Fix for Issue #475

### PR DESCRIPTION
**Pull request by mapete94**
[https://github.com/go-mgo/mgo/issues/475](https://github.com/go-mgo/mgo/issues/475])

### CODE REVIEW

File: session.go [hunk 1]
```diff
@@ -1537,8 +1537,21 @@ func simpleIndexKey(realKey bson.D) (key []string) {
 		field := realKey[i].Name
 		vi, ok := realKey[i].Value.(int)
```
Comment (niemeyer) [inline session.go]: This may be included in the type switch as well.

File: session.go [hunk 2]
```diff
@@ -1537,8 +1537,21 @@ func simpleIndexKey(realKey bson.D) (key []string) {
...
+			case float64:
+				vf, _ := realKey[i].Value.(float64)
+				vi = int(vf)
+			case string:
+				if vs, ok := realKey[i].Value.(string); ok {
+					key = append(key, "$"+vs+":"+field)
+					continue
+				}
+			default:
+				panic("Got unknown index key type for field " + field)
```
Comment (niemeyer) [inline session.go]: We pretty much never want to panic on bad data. The driver and the code is not at fault when the data is bogus.

**Review by niemeyer — CHANGES_REQUESTED** [review]
Thanks for the change. Some minor requests below, and a  test for this would be good as well.

**Comment by mapete94** [comment]
I made the requested changes. I did not add a test. I didn't see a way to create a 64bit int for your index type within the documentation. I may have missed it in the documentation, but if there is a way and I'm missing it I can write one.

File: session.go (hunk 1 above)
Comment (mapete94) [reply session.go]: It's been incorporated into the case statement.

File: session.go (hunk 2 above)
Comment (mapete94) [reply session.go]: I've removed it, but there is a similar panic right below it that was the reason I included it.

**Pull request merged**